### PR TITLE
Make ILiveListeners Algorithm Aware

### LINKED ISSUE #19319: Give ILiveListeners read-only Algorithm Access
This would allow LiveListeners to fetch properties off `StartLiveData` for example

### PR DESCRIPTION
**Pull request by OwenArnold**
Rather than having the LiveListenerFactories work with the ILiveListeners
as though they were just IPropertyManagers (updatePropertyValues). They
now call a specific `setAlgorithm` method. This currently just forwards to
`IPropertyManager::updatePropertyValues` for the `LiveListener` base class, but this can be overridden.

**To test:**

* Code review
* Ensure changes are fit for purpose
* Test sample listeners

Fixes #19319

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by OwenArnold** [comment]
@peterfpeterson I think the changes here are low risk, but It would be great if you could test out some of the SNS live listeners nevertheless as they could be affected by these changes.

File: Framework/API/src/LiveListener.cpp [hunk 1]
```diff
@@ -19,5 +20,14 @@ void LiveListener::setSpectra(const std::vector<specnum_t> &specList) {
   UNUSED_ARG(specList);
 }
 
+/**
+ * Default behaviour updates property values on Listener
+ * from those on calling algorithm.
+ */
+void LiveListener::setAlgorithm(const IAlgorithm &callingAlgorithm) {
+  this->updatePropertyValues(callingAlgorithm);
+  ;
```
Comment (matthew-d-jones) [inline Framework/API/src/LiveListener.cpp]: Stray semicolon?

**Review by matthew-d-jones — COMMENTED** [review]
:+1: 
This does what I want for the Kafka listener, I can override `setAlgorithm` to grab the value of `StartLiveData`'s `Instrument` property.

**Review by matthew-d-jones — APPROVED** [review]

**Review by AndreiSavici — APPROVED** [review]
Just code review: looks fine to me

**Comment by matthew-d-jones** [comment]
:shipit:

**Pull request merged**